api/src/test/java/org/openmrs/module/locationbasedaccess/aop/UserSearchAdviserTest.java

```diff
@@ -0,0 +1,617 @@
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+
+        List<Role> userRole = new ArrayList<Role>();
+        userRole.add(userService.getRole("userRole")); //Adding the role "userRole" to a new List
+        List<User> userList = userService.getUsers("User",userRole,true);
+        assertEquals(userList.size(), 4);
+        assertNotNull(userList.get(0));
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());
+        assertNotNull(userList.get(1));
+        assertEquals(userList.get(1).getUserProperty(DEMO_USER_PROPERTY),location2.getUuid());
+        assertNotNull(userList.get(2));
+        assertEquals(userList.get(2).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());
+        assertNotNull(userList.get(3));
+        assertEquals(userList.get(3).getUserProperty(DEMO_USER_PROPERTY),"");
+    }
+    
+    @Test
+    public void getUsers_nonAdminUserWithOutLocationPropertyShouldNotGetUserInformation(){
+        User normalUser = userService.getUser(DEMO_USER1_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        assertEquals(normalUser.getUserProperties().size(), 0);
+
+        List<Role> userRole = new ArrayList<Role>();
+        userRole.add(userService.getRole("userRole")); //Adding the role "userRole" to a new List
+        List<User> userList = userService.getUsers("User",userRole,true);
+        assertEquals(1, userList.size());
+        assertNotNull(userList.get(0)); 
+        assertEquals(userList.get(0),normalUser); //Users can access themselves
+    }
+    
+    @Test
+    public void getUsers_nonAdminUserWithLocationPropertyShouldGetOnlyAccessibleUserInformation(){
+        User normalUser = userService.getUser(DEMO_USER2_ID);
+        Context.authenticate(normalUser.getUsername(), "userServiceTest");
+        assertFalse(normalUser.isSuperUser());
+
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+
+        assertEquals(normalUser.getUserProperties().size(), 1);
+        String locationUserProperty = normalUser.getUserProperty("locationUuid");
+        assertNotNull(locationUserProperty);
+        assertEquals(locationUserProperty, location1.getUuid());
+
+        List<Role> userRole = new ArrayList<Role>();
+        userRole.add(userService.getRole("userRole")); //Adding the role "userRole" to a new List
+        List<User> userList = userService.getUsers("User",userRole,true);
+        assertEquals(userList.size(), 2);
+        assertNotNull(userList.get(0));
+        assertEquals(userList.get(0).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());
+        assertEquals(userList.get(0),normalUser); //Users can access themselves
+        assertNotNull(userList.get(1));
+        assertEquals(userList.get(1).getUserProperty(DEMO_USER_PROPERTY),location1.getUuid());

```

Comment:
After this line, Have you proof that the user object is not the normalUser? 
Because there can be some situations that both are normalUser. If both are normalUser, then your testing approach won't address your testing method requirements.

Reply:
I have added
assertNotEquals( userList.get(1), normalUser);
after line 133.